core/src/main/java/feast/core/service/JobCoordinatorService.java

```diff
@@ -148,27 +150,23 @@ public void Poll() throws InvalidProtocolBufferException {
       }
       completedTasks++;
     }
-
-    log.info("Updating feature set status");
-    updateFeatureSetStatuses(jobUpdateTasks);
+    executorService.shutdown();
   }
 
   // TODO: make this more efficient
   private void updateFeatureSetStatuses(List<JobUpdateTask> jobUpdateTasks) {
     Set<FeatureSet> ready = new HashSet<>();
```

Comment:
Not really related to this PR per say, but would it be possible for us to remove this `updateFeatureSetStatuses` call and use the real job state?

Reply:
I think you're asking: rather than cache feature set statuses in bulk, could status be returned on-demand by checking job state for one given feature set, when requested? I guess the team will have more context on the design.

As it stands, I'd have to get my head more into functional consideration of what it's doing beyond the mechanical refactoring I've done to this class, but it seems like there could be more to look at later:

1. In the `Poll()` loop we are:
   - At the beginning, doing `getJob` for the source/store combo of all feature sets that are subscribed to
   - Then, doing the `startOrUpdate` of tasks for all those, and waiting on all those tasks
   - Throwing away the results of waiting (updated `Job` references), because the `startOrUpdate` returns `void`
   - Doing `getJob` again for each of the tasks, in the `updateFeatureSetStatuses`

   It seems like this second round of DB lookups could be avoided by keeping and using the `Job`s we just got back. Unless I'm missing something, if it's trying to avoid a race condition or something.
2. `getJob` does `jobRepository.findBySourceIdAndStoreNameOrderByLastUpdatedDesc` returning all matching records, and then takes the first (most recent) one client-side instead of having SQL do it. And this happens in loops, 2x per polling loop as noted in 1.

Reply:
Tracking your comments here: #664 

> and then takes the first (most recent) one client-side instead of having SQL do it. 

That's a very low hanging fruit.